File: android/app/src/main/java/io/cpunk/dna/Wallet.java
package io.cpunk.dna;

public class Wallet {
    public static final int SIG_DILITHIUM = 0;
    public static final int SIG_PICNIC = 1;
    public static final int SIG_BLISS = 2;
    public static final int SIG_TESLA = 3;

    public final String name;
    public final String address;
    public final int sigType;
    public final boolean isProtected;

    public Wallet(String name, String address, int sigType, boolean isProtected) {
        this.name = name;
        this.address = address;
        this.sigType = sigType;
        this.isProtected = isProtected;
    }

    public String getSigTypeString() {
        switch (sigType) {
            case SIG_DILITHIUM: return "Dilithium";
            case SIG_PICNIC: return "Picnic";
            case SIG_BLISS: return "BLISS";
            case SIG_TESLA: return "TESLA";
            default: return "Unknown";
        }
    }
}

File: mobile/shared/src/commonMain/kotlin/io/cpunk/dna/domain/models/Wallet.kt
package io.cpunk.dna.domain.models

/**
 * Wallet data class
 * Represents a Cellframe wallet
 */
data class Wallet(
    val name: String,
    val address: String,
    val network: String,
    val sigType: WalletSignatureType,
    val isProtected: Boolean,
    val balances: Map<String, String> = emptyMap()  // Token -> Balance
)

enum class WalletSignatureType {
    DILITHIUM,
    PICNIC,
    BLISS,
    TESLA,
    UNKNOWN
}

/**
 * Transaction data class
 */
data class Transaction(
    val hash: String,
    val from: String,
    val to: String,
    val token: String,
    val amount: String,
    val fee: String,
    val timestamp: Long,
    val status: TransactionStatus
)

enum class TransactionStatus {
    PENDING,
    CONFIRMED,
    FAILED
}

/**
 * Token balance data class
 */
data class TokenBalance(
    val token: String,
    val balance: String,
    val network: String
)
